validator/utils/logging.py

```diff
@@ -1,76 +1,136 @@
 import json
+import logging
 from contextvars import ContextVar
 from logging import Formatter
 from logging import Logger
 from logging import LogRecord
 from logging.handlers import RotatingFileHandler
 from pathlib import Path
+from typing import Any
+from typing import Optional
 
-from fiber.logging_utils import get_logger
+from fiber.logging_utils import get_logger as fiber_get_logger
 
 
-def create_extra_log(**tags: str | None) -> dict[str, dict[str, str | None]]:
+current_context = ContextVar[dict[str, Any]]("current_context", default={})
+
+
+def add_context_tag(key: str, value: Any) -> None:
+    """Add or update a tag in the current logging context"""
     try:
-        context_task_id = current_task_id.get()
-        if context_task_id is not None:
-            tags["task_id"] = context_task_id
+        context = current_context.get()
+        new_context = {**context, key: value}
+        current_context.set(new_context)
+    except LookupError:
+        current_context.set({key: value})
+
+
+def remove_context_tag(key: str) -> None:
+    """Remove a tag from the current logging context"""
+    try:
+        context = current_context.get()
+        if key in context:
+            new_context = context.copy()
+            del new_context[key]
+            current_context.set(new_context)
     except LookupError:
         pass
-    return {"tags": {k: v for k, v in tags.items() if v is not None}}
+
+
+def clear_context() -> None:
+    """
+    Removes all tags from the current logging context.
+    """
+    current_context.set({})
+
+
+def get_context_tag(key: str) -> Optional[Any]:
+    """Get a tag value from the current logging context"""
+    try:
+        context = current_context.get()
+        return context.get(key)
+    except LookupError:
+        return None
 
 
 class JSONFormatter(Formatter):
     def format(self, record: LogRecord) -> str:
-        tags = record.__dict__.get("tags", {}) if hasattr(record, "tags") else {}
+        if not hasattr(record, "tags"):
+            try:
+                context = current_context.get()
+                record.tags = {k: v for k, v in context.items() if v is not None}
+            except LookupError:
+                record.tags = {}
+
         clean_level = record.levelname.replace("\u001b[32m", "").replace("\u001b[1m", "").replace("\u001b[0m", "")
-        log_data: dict[str, str | dict] = {
+        log_data: dict[str, Any] = {
```

Comment:
Namo isn't a fan of Any - do we truely mean Any or can we define some types we'd expect?

Reply:
hmm not sure what could be passed in. Ints maybe etc. But I can leave the str and dict and see if we need changing later